README.md

```diff
@@ -179,6 +180,58 @@ Output options:
 ```
 <!-- @end-sigstore-sign-help@ -->
 
+
+### Signing with DSSE envelopes
```

Comment:
Thinking out loud: we should probably add some documentation about when users might want to do this versus "normal" signing. But we can do that in a follow-up 🙂